src/sentry/api/endpoints/organization_events.py

```diff
@@ -315,8 +318,20 @@ def get(self, request: Request, organization) -> Response:
         if not self.has_feature(organization, request):
             return Response(status=404)
 
+        referrer = request.GET.get("referrer")
+
         try:
-            snuba_params, params = self.get_snuba_dataclass(request, organization)
+            # If the organization allows joining and leaving projects, it means that their
+            # developers can visit issues from any projects. In the case of trace related issues,
+            # we need the endpoint to let us see events from all projects when this flag is enabled.
+            snuba_params, params = self.get_snuba_dataclass(
+                request,
+                organization,
+                check_global_views=(
+                    referrer in GLOBAL_VIEW_WHITELIST
+                    and not bool(organization.flags.allow_joinleave)
```

Comment:
```suggestion
                # This is only temporary until we come to a decision on global views
                # checking for referrer for an allowlist is a brittle check since referrer
                # can easily be set by the caller
                check_global_views=(
                    referrer in GLOBAL_VIEW_WHITELIST
                    and not bool(organization.flags.allow_joinleave)
```